### PR TITLE
[SYCL] Add buffer dimensions restriction

### PR DESCRIPTION
**Pull request by MochalovaAn**
Adding a static_assert to the buffer dimensions.

Signed-off-by: amochalo <anastasiya.mochalova@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/buffer.hpp [hunk 1]
```diff
@@ -370,6 +370,10 @@ class buffer {
       return newRange[0] == 1 && newRange[2] == parentRange[2];
     return newRange[1] == parentRange[1] && newRange[2] == parentRange[2];
   }
+
+  static_assert (dimensions > 0 && dimensions <= 3,
+		  "Dimensions can be 1/2/3 for buffer.");
```
Comment (asavonic) [inline sycl/include/CL/sycl/buffer.hpp]: Range class has a similar assert, so I suggest to use the same wording here:
> range can only be 1, 2, or 3 dimensional

File: sycl/include/CL/sycl/buffer.hpp [hunk 2]
```diff
@@ -370,6 +370,10 @@ class buffer {
       return newRange[0] == 1 && newRange[2] == parentRange[2];
     return newRange[1] == parentRange[1] && newRange[2] == parentRange[2];
   }
+
+  static_assert (dimensions > 0 && dimensions <= 3,
+		  "Range can only be 1/2/3 dimensional.");
```
Comment (Fznamznon) [inline sycl/include/CL/sycl/buffer.hpp]: I suggest mention buffer in diagnosing of buffer usage:
```suggestion
		  "Range of the buffer can only be 1/2/3 dimensional.");
```

File: sycl/include/CL/sycl/buffer.hpp [hunk 3]
```diff
@@ -370,6 +370,10 @@ class buffer {
       return newRange[0] == 1 && newRange[2] == parentRange[2];
     return newRange[1] == parentRange[1] && newRange[2] == parentRange[2];
   }
+
+  static_assert (dimensions > 0 && dimensions <= 3,
+		  "Range of the buffer can only be 1/2/3 dimensional.");
+
```
Comment (AGindinson) [inline sycl/include/CL/sycl/buffer.hpp]: I believe `clang-format` should applied for the commit.

File: sycl/include/CL/sycl/buffer.hpp [hunk 4]
```diff
@@ -370,6 +370,10 @@ class buffer {
       return newRange[0] == 1 && newRange[2] == parentRange[2];
     return newRange[1] == parentRange[1] && newRange[2] == parentRange[2];
   }
+
+  static_assert (dimensions > 0 && dimensions <= 3,
+		  "Range of the buffer can only be 1/2/3 dimensional.");
```
Comment (bader) [inline sycl/include/CL/sycl/buffer.hpp]:
```suggestion
		  "A SYCL buffer can only be 1/2/3 dimensional.");
```
What does "Range of the buffer" mean?

**Pull request closed without merging**